contracts/HubPool.sol

```diff
@@ -241,11 +236,12 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         uint256 lpTokenAmount,
         bool sendEth
     ) public nonReentrant {
-        // Can only send eth on withdrawing liquidity iff this is the WETH pool.
         require(pooledTokens[l1Token].isWeth || !sendEth, "Cant send eth");
         uint256 l1TokensToReturn = (lpTokenAmount * _exchangeRateCurrent(l1Token)) / 1e18;
 
         ExpandedIERC20(pooledTokens[l1Token].lpToken).burnFrom(msg.sender, lpTokenAmount);
+        // Note this method does not make any liquidity utilization checks before letting the LP redeem their LP tokens.
+        // If they try access more funds that available (i.e l1TokensToReturn > liquidReserves) this will underflow.
```

Comment:
this underflow is expected + fine in this case right?

Reply:
yup! solidity 8 protects against underflows so this simply reverts.

Reply:
Is it possible for this to underflow, but the contract to have enough tokens to transfer out?